### PR TITLE
onfocus() must not be called on disabled input

### PR DESCRIPTION
**Pull request by Farfurix**
References: 
https://github.com/DevExpress/testcafe/issues/2123, 
https://github.com/DevExpress/testcafe/issues/2037

### CODE REVIEW

File: test/client/fixtures/sandbox/event/focus-blur-change-test.js [hunk 1]
```diff
@@ -261,6 +261,22 @@ asyncTest('without handlers', function () {
     testFocusing(0, startNext);
 });
 
+asyncTest('onfocus() must not be called on disabled input', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/event/focus-blur-change-test.js]: Rename to `asyncTest('disabled element', function () {`

File: test/client/fixtures/sandbox/event/focus-blur-change-test.js [hunk 2]
```diff
@@ -261,6 +261,22 @@ asyncTest('without handlers', function () {
...
+    var focused = false;
+    var onfocus = function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/event/focus-blur-change-test.js]: Also there is a non bubbling equivalent of `focus` event - `focusin`.
Check a native browser behavior:   Does disable input raise a `focusin` event?

Comment (Farfurix) [reply test/client/fixtures/sandbox/event/focus-blur-change-test.js]: Yes, these changes also avoid "onfocusin" handler calling. Example:
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<form id="parent-form">
    <input id="child-input" value="input">
</form>
<script>
    var parentForm      = document.getElementById('parent-form');
    var childInput      = document.getElementById('child-input');
    childInput.disabled = true;

    parentForm.addEventListener('focusin', function () {
        console.log('focusin');
    });

    childInput.focus();
</script>
</body>
</html>
```

File: test/client/fixtures/sandbox/event/focus-blur-change-test.js (hunk 1 above)
Comment (Farfurix) [reply test/client/fixtures/sandbox/event/focus-blur-change-test.js]: done

**Comment by AlexanderMoskovkin** [comment]
This should fix one more issue: https://github.com/DevExpress/testcafe/issues/2037

**Comment by AlexanderMoskovkin** [comment]
Please close them both in the TestCafe repo when you will update the `testcafe-hammerhead` version there

**Comment by miherlosev** [comment]
@AlexanderMoskovkin review too

File: src/client/sandbox/event/focus-blur.js [hunk 3]
```diff
@@ -233,6 +233,9 @@ export default class FocusBlurSandbox extends SandboxBase {
     }
 
     focus (el, callback, silent, forMouseEvent, isNativeFocus, preventScrolling) {
+        if (el.disabled)
+            return null;
+
```
Comment (AlexanderMoskovkin) [inline src/client/sandbox/event/focus-blur.js]: lgtm.

But before merge please recheck whether we need the `this.shouldDisableOuterFocusHandlers` property check below. I can't find any place where we set it to `true`

Comment (miherlosev) [reply src/client/sandbox/event/focus-blur.js]: Remove the flag `this.shouldDisableOuterFocusHandlers` and whole condition
```js
if (this.shouldDisableOuterFocusHandlers && !domUtils.isShadowUIElement(el))
            return null;
```
It never used in `testcafe`, `testcafe-hammerhead`, `testcafe-studio` and `testcafe-legacy-api` repositories.

Comment (Farfurix) [reply src/client/sandbox/event/focus-blur.js]: Done.

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
ping @AlexanderMoskovkin

**Pull request merged**